apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -105,8 +110,38 @@ class VisualizerModal extends React.Component {
     return columns.filter(column => isColumnNumeric(records, column));
   });
 
+  getValuesForFilterColumn = memoize((records, column) => {
+    if (!records || !column) {
+      return [];
+    }
+    let values = Array.from(new Set(records.map(record => record[column])));
+    return sortAlphabeticallyOrNumerically(values);
+  });
+
+  filterRecords = memoize((records, column, value) => {
+    if (!records) {
+      return [];
+    }
```

Comment:
This might express more nicely as a default parameter, assuming the only non-value you're expecting is `undefined`.
```js
filterRecords = memoize((records = [], column, value) => {
```

Reply:
Ah thanks that's definitely cleaner! I *think* `records` will always be defined anyways, so this check is probably overly cautious.